docs/RFCS/system_jobs.md

```diff
@@ -0,0 +1,421 @@
+query, is planned.
+
+## Minor considerations
+
+- Cloud storage credentials can be passed in the query for `BACKUP` and
+  `RESTORE` commands. Care will be taken to strip those credentials from the
+  query text before the text is inserted into the `system.jobs` table.
+
+- Unlike past proposals, the schema intentionally lacks a `name` or
+  `description` column. As it stands, none of the job-creating queries allow the
+  user to specify such a name or description.
+
+# Drawbacks
+
+Requiring the job leader to periodically issue `UPDATE system.jobs SET
+percentCompleted = ...` queries to update the progress of running jobs is
+somewhat unsatisfying. One wishes to be able to conjure the `percentCompleted`
+column only when the record is read, but this design introduces significant
+implementation complexity.
+
+# Alternatives
+
+## Wider protobufs
+
+To further minimize the chances that we'll need to modify the `system.jobs`
+schema, we could instead stuff all the data into the `payload` protobuf:
+
+```sql
+CREATE TABLE system.jobs (
+    id      INT PRIMARY KEY,
+    jobType STRING NOT NULL,
+    payload BYTES,
+)
+```
+
+This allows for complete flexibility in adjusting the schema, but prevents
+essentially all useful SQL queries over the table until protobuf columns are
+natively supported.
+
+## Narrow protobufs
+
+We could also allow all data to be filtered by widening the `system.jobs` table
+to include all the fields proposed to be stored in the `payload` protobuf,
+potentially dropping `payload` entirely.
+
+```sql
+CREATE TABLE system.jobs (
+    id                INT DEFAULT unique_rowid() PRIMARY KEY,
+    status            STRING NOT NULL,
+    percentCompleted  DECIMAL(5, 2),
+    payload           BYTES,
+    query             STRING,
+    creator           STRING,
+    created           TIMESTAMP NOT NULL,
+    started           TIMESTAMP,
+    lastUpdated       TIMESTAMP NOT NOULL,
+    error             STRING,
+    backup_to         STRING,
+    backup_targets    STRING,
+    restore_from      STRING,
+    schema_change_tbl STRING,
+    INDEX status_idx  (status)
+)
+```
+
+This option allows the table to be indexed on any column or queried on any
+column with SQL, but poses a significant risk if we need to adjust the schema.
+This design does unlock some useful SQL queries, however. Retrieving the second
+page of completed backup jobs can be expressed very naturally, for example:
+
+```sql
+SELECT * FROM system.jobs
+  WHERE jobType = 'backup'
+        status IN ('succeeded', 'failed')
+  ORDER BY created
+  LIMIT 50
+  OFFSET 50
+```
+
+## More protobufs
+
+Orthogonal to the above concern is whether we separate common fields from
+job-specific fields in the protobuf. For example, with wide protobufs, the table
+schema would look like this
+
+```sql
+CREATE TABLE system.jobs (
+    id             INT PRIMARY KEY,
+    jobType        STRING NOT NULL,
+    jobPayload     BYTES,
+    detailsPayload BYTES,
```

Comment:
I would delete this whole alternative (or shorten it). it feels like a lot of text for something that doesn't get any any particular advantages